utils/appliance/__init__.py

```diff
@@ -1646,31 +1619,36 @@ def is_idle(self):
 
     @cached_property
     def build_datetime(self):
-        datetime = self.ssh_client.get_build_datetime()
-        return datetime
+        build_datetime_string = self.build.split('_', 1)[0]
+        return datetime.strptime(build_datetime_string, '%Y%m%d%H%M%S')
 
     @cached_property
     def build_date(self):
-        date = self.ssh_client.get_build_date()
-        return date
-
-    @cached_property
-    def is_downstream(self):
-        return self.ssh_client.is_appliance_downstream()
+        return self.build_datetime.date()
 
     def has_netapp(self):
         return self.ssh_client.appliance_has_netapp()
 
     @cached_property
     def guid(self):
-        result = self.ssh_client.run_command('cat /var/www/miq/vmdb/GUID')
-        return result.output
+        try:
+            server = self.rest_api.get_entity_by_href(self.rest_api.server_info['server_href'])
+            return server.guid
+        except (AttributeError, KeyError, IOError):
+            self.log.exception('appliance.guid could not be retrieved from REST, falling back')
+            result = self.ssh_client.run_command('cat /var/www/miq/vmdb/GUID')
+            return result.output
 
     @cached_property
     def evm_id(self):
-        miq_servers = self.db.client['miq_servers']
-        return self.db.client.session.query(
-            miq_servers.id).filter(miq_servers.guid == self.guid)[0][0]
+        try:
+            server = self.rest_api.get_entity_by_href(self.rest_api.server_info['server_href'])
```

Comment:
this piece seems to be duplicated a lot, perhaps a helper can cut the code short

Reply:
Yeh, server_info api fields could have a getter method on appliance.